http/grafana.go

```diff
@@ -0,0 +1,288 @@
+			for rowIndex := 0; rowIndex < values.Len(); rowIndex++ {
+				if len(retVal.Rows) <= rowIndex {
+					retVal.Rows = append(retVal.Rows, []interface{}{})
+				}
+				value := values.Index(rowIndex).Interface()
+				retVal.Rows[rowIndex] = append(retVal.Rows[rowIndex], value)
+			}
+		}
+	}
+	if iter.Err() != nil {
+		return nil, iter.Err()
+	}
+	return []TableOutput{retVal}, nil
+}
+
+func (req *simpleJsonQueryRequest) formatTSDB(iter frames.FrameIterator) (interface{}, error) {
+	retval := []TimeSeriesOutput{}
+	for iter.Next() {
+		frame := iter.At()
+		iface, ok := frame.(pb.Framed)
+		if !ok {
+			return nil, errors.New("unknown frame type")
+		}
+		target := req.getTargetTSDB(iface.Proto())
+		indices := frame.Indices()
+		for _, column := range iface.Proto().Columns {
+			if column.Name == "values" {
+				datapoints := [][]interface{}{}
+				values, _ := req.extractColumnValues(column)
+				times, err := indices[0].Times()
+				if err != nil {
+					return nil, err
+				}
+				for j := 0; j < values.Len(); j++ {
+					datapoints = append(datapoints, []interface{}{values.Index(j).Interface(), req.formatTimeTSDB(times[j])})
+				}
+				retval = append(retval, TimeSeriesOutput{datapoints, target})
+			}
+		}
+	}
+	if iter.Err() != nil {
+		return nil, iter.Err()
+	}
+	return retval, nil
+}
+
+func (req *simpleJsonQueryRequest) extractColumnValues(column *pb.Column) (reflect.Value, string) {
+	colTypeStr := strings.ToLower(column.Dtype.String())
+	var dataSlice interface{}
+	switch column.Dtype {
+	case pb.DType_INTEGER:
+		dataSlice = &column.Ints
+		break
+	case pb.DType_FLOAT:
+		dataSlice = &column.Floats
+		break
+	case pb.DType_STRING:
+		dataSlice = &column.Strings
+		break
+	case pb.DType_TIME:
+		dataSlice = &column.Times
+		break
+	case pb.DType_BOOLEAN:
+		dataSlice = &column.Bools
+		break
+	}
+	return reflect.ValueOf(dataSlice).Elem(), colTypeStr
+}
+
+func (req *simpleJsonQueryRequest) formatTimeTSDB(timestamp time.Time) interface{} {
+	return timestamp.UnixNano() / 1000000
+}
+
+func (req *simpleJsonQueryRequest) getTargetTSDB(frame *pb.Frame) string {
+	labels := ""
+	metricName := ""
+	for _, column := range frame.Columns {
+		if column.Name != "values" {
+			values, _ := req.extractColumnValues(column)
+			val := fmt.Sprintf("%s", values.Index(0).Interface())
+			if column.Name == "metric_name" {
+				metricName = val
+			} else {
+				labels += fmt.Sprintf("[%s=%s]", column.Name, val)
+			}
+		}
+	}
+	return metricName + labels
+}
+
+func (req *simpleJsonQueryRequest) CreateResponse(iter frames.FrameIterator) (interface{}, error) {
+	formatters := map[string]func(iter frames.FrameIterator) (interface{}, error){"kv": req.formatKV, "tsdb": req.formatTSDB}
+	if formatter, ok := formatters[req.Backend]; ok {
+		return formatter(iter)
+	}
+	return nil, fmt.Errorf("Unknown format: %s", req.Backend)
+}
+
+func (req *simpleJsonSearchRequest) CreateResponse(iter frames.FrameIterator) (interface{}, error) {
```

Comment:
This is one reaon why I don't like the generic approach with `simpleJsonRequestInterface`. It is used just on one place and here it's not supported - [YAGNI](https://martinfowler.com/bliki/Yagni.html)